servers/services/src/main/java/org/projectnessie/services/rest/BaseResource.java

```diff
@@ -59,14 +62,44 @@ protected BaseResource(
     }
   }
 
-  WithHash<Ref> namedRefWithHashOrThrow(String ref) throws NessieNotFoundException {
+  WithHash<NamedRef> namedRefWithHashOrThrow(String namedRef, @Nullable String hashOnRef)
+      throws NessieNotFoundException {
+    List<WithHash<NamedRef>> collect =
+        store
```

Comment:
how often is this called? This could be expensive and I don't quite understand the purpose

Reply:
it's called in 4 places (getCommitLog / getEntries / getContents / getMultipleContents) and we will eventually have a faster alternative in in the VersionStoreV2